### PR TITLE
Split Activation Fit function into two seperate functions

### LINKED ISSUE #33090: Split Activation Fitting function
**Is your feature request related to a problem? Please describe.**
At present the Activation Fit Function requires users to input a suitable string for the Unit as it calculates differently depending on whether its K or meV. However the user is unhappy with this solution as inputting an invalid string will only give an error once the Fit button has been clicked (this is because it is a fitting function not a GUI so input cannot be checked prior to this). Whilst the documentation is clear about which units can be used and the resulting error message also informs users which they should use we have been asked to come up with an alternative way of providing this fitting function.

It should be noted that the function may also need to be expanded to include other units.

**Describe the solution you'd like**
To use a helper function (see #33085 ) that provides the necessary calculations to two separate fitting functions, one for K and one for meV. This ensures the user chooses the correct function from the start and reduces the number of parameters they need to update.

**Describe alternatives you've considered**
It is not possible to use a drop down menu for selecting units because this is a Fitting Function not a GUI. 

The user suggested using a code ( 1 = K and 2 = meV) and somehow prevent the user from inputting any other numbers. However this could lead to confusion as users would have to know which number to use for which unit. This would have to be included in the documentation but the original solution also relied on users making use of the documentation which didn't work.

**Additional context**
PR for the Activation Fit Function - #32899

### PR DESCRIPTION
**Pull request by sf1919**
**Description of work.**
Split the Activation Fitting function for Muons into two separate functions, one for Kelvin and one for meV.

**To test:**

1. Enable the model analysis tab and plot (see #32572 for details on how to do this)
3. Open Muon Analysis GUI
4. Change Instrument to EMU
5. Load runs 20901-18
6. Go to the Grouping Tab
7. Guess Alpha
8. Go to fitting tab and add a DynamicKuboToyabe Fitting Function
9. Fix the Field to 0, and set Delta to 0.35 and Nu to 0.01
10. Go to the Sequential Fitting Tab and select runs 20901-17 and then sequentially fit selected
11. Go to the Results Tab
12. Select Temp_Sample from the log results and then click Output Results
13. Open the Model fitting tab and change x axis to Temp_sample and y axis to Nu
14. Add an ActivationK Fit Function and fit (leave Unit as K). AttemptRate = 33.4 +- 3.6 us^-1 and Barrier = 589.7+-13.9 and ChiSquared should be about 3.75 or better.

Note:
We have not been provided with data in meV to be able to directly test ActivationmeV fitting function. However as the conversion from Kelvin to meV is provided by https://github.com/mantidproject/mantid/blob/15ef1d68a3427cf52dbf91faa2072ca5c5bae02d/Framework/Kernel/inc/MantidKernel/PhysicalConstants.h#L63 which is already tested and the above tests the other part of the functionality ActivationmeV uses then provided the above functional test works, the code is reviewed and all the CI tests pass then ActivationmeV should be good to go.

Fixes #33090. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by SilkeSchomann — DISMISSED** [review]
Code looks good and the ActivationK Fitting function works as expected.

**Review by DavidFair — CHANGES_REQUESTED** [review]
I'm happy with the code, but we need to be sure that this is not a breaking change in any way.

Does 6.2 have Activation as a function? Since `FitScriptGenerator` will have potentially created scripts with this in now

File: Framework/CurveFitting/inc/MantidCurveFitting/Functions/ActivationK.h [hunk 1]
```diff
@@ -30,21 +25,17 @@ Activation parameters:
 </UL>
 */
 
-class MANTID_CURVEFITTING_DLL Activation : public BackgroundFunction {
+class MANTID_CURVEFITTING_DLL ActivationK : public BackgroundFunction {
```
[the remark this answers is not in the document]
Comment (sf1919) [reply Framework/CurveFitting/inc/MantidCurveFitting/Functions/ActivationK.h:28]: I'd be very surprised if it broke any exisiting scripts etc. Activation Fitting function was only added a few weeks ago (#32899) and as far as I'm aware only one user has tried to use it. These changes are as a result of their feedback (#33090).

**Review by SilkeSchomann — APPROVED** [review]
Approved again after recent changes.

**Review by DavidFair — APPROVED** [review]

**Pull request merged**